app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-page-views.js

```diff
@@ -9,13 +9,15 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
     PIIRemover: new window.GOVUK.analyticsGa4.PIIRemover(), // imported in analytics-ga4.js
     nullValue: undefined,
 
-    init: function () {
+    init: function (dynamicPageViewData) {
+      dynamicPageViewData = dynamicPageViewData || {}
+
       if (window.dataLayer) {
         var data = {
           event: 'page_view',
           page_view: {
             location: this.getLocation(),
-            referrer: this.getReferrer(),
+            referrer: dynamicPageViewData.referrer || this.getReferrer(),
```

Comment:
It looks like we're only interested in the `dynamicPageViewData` value for `referrer`? If that's the case do we need to pass a whole object or just this value?

Reply:
A little later in the code I also need to access another property (`dynamicPageViewData.dynamic`) so that the PA’s have something to differentiate dynamic page updates

Reply:
Yes, that's what I'm getting at. I'll resolve this comment though as the other thread is talking about the same thing.